File: solutions/java/log-levels/1/src/main/java/LogLevels.java
import java.io.*;
import java.util.*;
public class LogLevels {
    
    public static String message(String logLine) {
           String[] info = logLine.split(":");
            info[1] = info[1].trim();
             return info[1];
           
    }

    public static String logLevel(String logLine) {
      String[] info = logLine.split(":");
            info[0] = info[0].replaceAll("[\\[\\]]", "");
            info[0] = info[0].toLowerCase();
             return info[0];
    }

    public static String reformat(String logLine) {
        String[] info = logLine.split(":");
        info[0] = info[0].replaceAll("[\\[]", "(");
        info[0] = info[0].replaceAll("[]//]", ")");
        info[0] = info[0].toLowerCase();
        info[0] = info[0].trim();
        info[1] = info[1].trim();
        return info[1] + " " +info[0];
    }
}

File: solutions/cpp/log-levels/1/log_levels.cpp
#include <string>
using std::string;
namespace log_line {
string message (string log_message){
    int postion = log_message.find(":") ;
    string display =  log_message.substr(postion + 2, 22);
    return display;
}
string log_level(string log_message){
    int postion = log_message.find(":") ;
    string display =  log_message.substr(1,postion-2);
    return display;
}
string reformat(string log_message){
    return (message(log_message) + " (" + log_level(log_message) +")");
}

} // namespace log_line
